From aa5f0cf860c8aa4adc3bf67071bae9390de68ac6 Mon Sep 17 00:00:00 2001
From: Elaine Zhang <zhangqing@rock-chips.com>
Date: Thu, 14 Apr 2016 05:48:47 +0800
Subject: [PATCH] ARM64: dts: rockchip: rk808: set the dvs2 gpio pull down

the hw default of the dvs2 is pull up which is not correct.
set the dvs2 gpio pull down.

Change-Id: I0d296cecc422456cb72630d5ce64a5c7e5dad283
Signed-off-by: Elaine Zhang <zhangqing@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi | 14 +++++++++++++-
 arch/arm64/boot/dts/rockchip/rk3399.dtsi    |  7 -------
 2 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi
index 889a31968324..8e278880abb4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-tb.dtsi
@@ -217,6 +217,18 @@
 			rockchip,pins = <0 10 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
+
+	pmic {
+		pmic_int_l: pmic-int-l {
+			rockchip,pins =
+				<1 21 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+
+		pmic_dvs2: pmic-dvs2 {
+			rockchip,pins =
+				<1 18 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
+	};
 };
 
 &sdmmc {
@@ -302,7 +314,7 @@
 		interrupt-parent = <&gpio1>;
 		interrupts = <21 IRQ_TYPE_LEVEL_LOW>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&pmic_int_l>;
+		pinctrl-0 = <&pmic_int_l &pmic_dvs2>;
 		rockchip,system-power-controller;
 		wakeup-source;
 		#clock-cells = <1>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index 417dacea7f98..20d8bf83f1df 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -1777,13 +1777,6 @@
 			};
 		};
 
-		pmic {
-			pmic_int_l: pmic-int-l {
-				rockchip,pins =
-					<1 21 RK_FUNC_GPIO &pcfg_pull_up>;
-			};
-		};
-
 		edp {
 			edp_hpd: edp-hpd {
 				rockchip,pins =
-- 
2.35.3

